lib/core/src/swapper/boltz/mod.rs

```diff
@@ -147,23 +147,36 @@ impl<P: ProxyUrlFetcher> BoltzSwapper<P> {
         Ok(bitcoin_client)
     }
 
+    /// Get the partial signature and pub nonce for a chain swap.
+    /// If we fail to get the claim tx details, we may have already sent it, so we return None.
     async fn get_claim_partial_sig(
         &self,
         swap: &ChainSwap,
-    ) -> Result<(MusigPartialSignature, MusigPubNonce), PaymentError> {
+    ) -> Result<Option<(MusigPartialSignature, MusigPubNonce)>, PaymentError> {
         let refund_keypair = swap.get_refund_keypair()?;
 
         // Create a temporary refund tx to an address from the swap lockup chain
         // We need it to calculate the musig partial sig for the claim tx from the other chain
         let lockup_address = &swap.lockup_address;
 
-        let claim_tx_details = self
+        let claim_tx_details = match self
             .get_boltz_client()
             .await?
             .inner
             .get_chain_claim_tx_details(&swap.id)
-            .await?;
-        match swap.direction {
+            .await
+        {
+            Ok(claim_tx_details) => claim_tx_details,
+            Err(boltz_client::error::Error::JSON(e)) => {
+                warn!("Failed to parse chain claim tx details: {e} - continuing without signature as we may have already sent it");
+                return Ok(None);
```

Comment:
What actuall error do we get here from boltz backend? Is it something we can use to return a structure error (like SignatureAlreadySent) ?